src/languageSupport/languages/flux/lsp/worker/flux.worker.ts

```diff
@@ -1,4 +1,89 @@
-import Buffer from 'src/languageSupport/languages/flux/lsp/worker/buffer'
+import {Lsp} from '@influxdata/flux-lsp-browser'
+import {Methods} from 'src/languageSupport/languages/flux/lsp/utils'
+
+const respond = (msg, cb) => {
+  try {
+    // LSP spec permits a blank response
+    if (!msg) {
+      return
+    }
+    const d = JSON.parse(msg)
+    cb(d)
+  } catch (e) {
+    console.error(e)
+  }
+}
+
+class Buffer {
+  private _i = 0
+  private _o = 0
+  private _initBuffer = new Array(100)
+  private _serverStartSignaled = false
+  private _initialized = false // ACK received from monaco
+  private _server: Lsp
+
+  private _insertBuffer = req => {
+    this._initBuffer[this._i % 100] = req
```

Comment:
You know, I'm not really sure. The only thing I did was move another file's contents into this file, because it was only being used here. I understand why this `Buffer` interface was created, but there's a more correct way to handle this. [There's an issue open currently to remove it](https://github.com/influxdata/ui/issues/5859).

Reply:
It's a ring buffer. One classic use case is handling incoming messages while waiting for something to boot up, (in this case the LSP server).

Since the original goals included having no change in the boot/load time with the LSP server worker, (a.k.a. the full UI page load time until fully functional), we were loading the monaco-editor in the UI (with it's own worker), in parallel with booting the LSP server (on the worker thread we made). Since the LSP server takes much longer to boot --> the ring buffer holds the incoming messages until the LSP server is up and running.

The new approach Paul is using, does not include a restriction on the page load/boot time. The bits are booted in series, with the LSP first, which means no incoming messages (to the LSP) needs to be buffered.

Reply:
Circular queuing. It goes index to length --> then start over again at index 0. Modulo does that.